java/core/src/java/org/apache/orc/impl/InStream.java

```diff
@@ -558,9 +558,14 @@ public void changeIv(Consumer<byte[]> modifier) {
     public void seek(PositionProvider index) throws IOException {
       boolean seeked = seek(index.getNext());
       long uncompressedBytes = index.getNext();
-      if (!seeked && uncompressed != null) {
-        // Only reposition uncompressed
-        uncompressed.position((int) uncompressedBytes);
+      if (!seeked) {
+        if (uncompressed != null) {
+          // Only reposition uncompressed
+          uncompressed.position((int) uncompressedBytes);
+        } else {
```

Comment:
I'd drop the empty else block, but the comment is good.

Reply:
Sure, dropped the empty block